File: Local/etc/modulefiles/lua/core/anaconda/2.7.lua
family('python')

-- Determine platform specific prefix
local uname = string.lower(capture("uname -a"))
local darwin = uname:find('darwin')
if darwin then
  root = '/opt'
else
  home = os.getenv('HOME')
  root = pathJoin(home, '.swx')
end
local name = myModuleName()
local version = myModuleVersion()
prefix = pathJoin(root, 'apps', name, version)

if (not isDir(prefix)) then
  LmodError('Load Error: ', prefix, 'does not exist')
end

local PATH = pathJoin(prefix, 'bin')
if isDir(PATH) then
  prepend_path('PATH', PATH)
end

local filename = pathJoin(prefix, 'lib/python' .. version,
                          'site-packages/spyderlib/spyder.py')
if isFile(filename) then
  set_alias('spyder', 'python ' .. filename)
end

File: Local/etc/modulefiles/core/anaconda/2.7.py
family('python')

# Determine platform specific prefix
darwin = system_is_darwin()
if darwin:
    root = '/opt'
else:
    home = getenv('HOME')
    root = os.path.join(home, '.swx')

name = self.name
version = self.version
prefix = os.path.join(root, 'apps', name, version)

if not os.path.isdir(prefix):
    log_error('Load Error: ' + prefix + ' does not exist')

PATH = os.path.join(prefix, 'bin')
if os.path.isdir(PATH):
    prepend_path('PATH', PATH)

filename = os.path.join(prefix, 'lib/python' + version,
                        'site-packages/spyderlib/spyder.py')
if os.path.isfile(filename):
    set_alias('spyder', 'python ' + filename)
